test/compilable/interpret3.d

```diff
@@ -897,13 +897,6 @@ int staticdynamic()
 }
 static assert(staticdynamic() == 0);
 
-int[] crashing()
-{
-    int[12] cra;
-    return (cra[2 .. $] = 3);
-}
-static assert(crashing()[9] == 3);
```

Comment:
We had that discussion a couple of times already, and you did understood the argument the last time.
It doesn't matter whether the code is ugly, broken, incorrect, or reverse. You don't know it's role in D projects, it might never be executed, or it might just work out of chance, or it's a memory corruption that by chance causes no harms.
It might even affect a crucial system which can tolerate failures and retries until it succeeds.

Unless it's prohibitively complex to avoid breakage, you should not make random decisions on what is correct or wrong, because you're not the one to fix the million lines of D code out there.

> Allowing this kind of code in the name of backwards compatibility is a disservice to our users.

Nonsense, our service for D users is to transition to a safer language, so that such errors are detected and prevented, and doing that in a way that is predictable for people (e.g. a "6 month to fix that" deprecation).
Here you just break stuff completely unscheduled and with no idea about the consequences.